### PR TITLE
Add helm integration test with otlp sink deployment

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
These changes add a helm chart metric* integration test case, a corresponding helm chart installer helper using that library's exported actions, and an "otlp-sink" deployment that mirrors the o11y/cloud backends but forwards all telemetry to the validating otlp sink. It also includes a logging mock o11y api that can be updated in the future to validate dimension updates.

The target helm chart path is configurable by env var, and defaults to a decremented minor version based on this project's (e.g. a `git describe HEAD` of `v0.87.0-22-g6eb7e6d1` will test v0.86.0 of the helm chart).

Though a large PR, the bulk of the changes are test yaml content and go.mod changes, including a 3300+ line resource metrics file for all the metrics sent by a (near) default chart installation.

### CODE REVIEW

File: tests/testutils/kubeutils/otlp_sink_deployment.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	return deployment
+}
+
+func (dep OTLPSinkDeployment) Teardown() {
+	dep.sb.shutdown()
+	so, se, err := dep.cluster.Delete(dep.manifests())
+	require.NoError(dep.tc, err, "stdout: %s, stderr: %s", so.String(), se.String())
+}
+
+func (dep OTLPSinkDeployment) apply() {
```
Comment (rmfitzpatrick) [inline tests/testutils/kubeutils/otlp_sink_deployment.go:72]: @jinja2 (re: https://github.com/signalfx/splunk-otel-collector/pull/3873#discussion_r1378000551) fwiw I count 163 lines from here to the end of the manifest related bits of this deployment, including the actual `kubectl apply`. The manifests alone in the test file are 165 lines and don't benefit from types, formatting, etc.

Comment (rmfitzpatrick) [reply tests/testutils/kubeutils/otlp_sink_deployment.go:72]: to be fair, applying the manifests would only add a line or two, so this isn't worth much.

**Comment by jvoravong** [comment]
The changes LGTM.

For future consideration, we should think about regularly updating the version of the kind cluster ([currently 1.26](https://github.com/signalfx/splunk-otel-collector/blob/9db038d1891ede4cf41efbbe2eb3195e846e87a6/tests/testutils/kubeutils/kind_cluster.go#L51C6-L51C6)) used in the collector helm integration tests. This version can influence the metrics generated during testing. The versions used for chart functional tests are specified [here](https://github.com/signalfx/splunk-otel-collector-chart/blob/beb88a295765393f6ddd156c7b88a9d8249d3e15/.github/workflows/functional_test_v2.yaml#L21). It might be beneficial to align the collector helm integration tests with one of the versions (preferably the latest) used in the chart functional tests in some manner.

**Review by jvoravong — APPROVED** [review]

**Comment by rmfitzpatrick** [comment]
> For future consideration, we should think about regularly updating the version of the kind cluster ([currently 1.26](https://github.com/signalfx/splunk-otel-collector/blob/9db038d1891ede4cf41efbbe2eb3195e846e87a6/tests/testutils/kubeutils/kind_cluster.go#L51C6-L51C6)) used in the collector helm integration tests. This version can influence the metrics generated during testing. The versions used for chart functional tests are specified [here](https://github.com/signalfx/splunk-otel-collector-chart/blob/beb88a295765393f6ddd156c7b88a9d8249d3e15/.github/workflows/functional_test_v2.yaml#L21). It might be beneficial to align the collector helm integration tests with one of the versions (preferably the latest) used in the chart functional tests in some manner.

Having consensus on which project should test which permutations/features would be helpful. IMO our collector distro tests should include all telemetry types for a most a couple version of k8s, which more comprehensive assurances enforced in the helm chart project, as the actual k8s resource provider. The tests in the collector project(s) should really focus on surfacing clear blockers from component changes that would otherwise only be apparent when updating the app version in the helm chart.

**Pull request merged**